### PR TITLE
make auth step, integrate jw3t, abstract auth

### LINKED ISSUE #1087: UI: components for fabric (steps) - Investor onboarding flow integration (v1)
### Acceptance Criteria

Create Stepper component in Fabric library

### LINKED ISSUE #1089: UI: set up auth - Investor onboarding flow integration (v1)
* make jw3t auth generic - from POD implementation (talk to Onno)

### PR DESCRIPTION
**Pull request by jpangelle**
### Description

This pull request:

- abstracts the `signToken` from the `pod` module in `centrifuge-js` to an `auth` module
- integrates an auth step into the onboarding flow
- migrates `Stepper` component to fabric
- UI tweaks

Closes #1087 
Closes #1089

### Approvals

- [ ] Dev
- [ ] Product

### CODE REVIEW

File: centrifuge-app/src/pages/Onboarding/LinkWallet.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { Box, Button, IconCheck, Shelf, Stack, Text } from '@centrifuge/fabric'
```
Comment (jpangelle) [inline centrifuge-app/src/pages/Onboarding/LinkWallet.tsx:1]: component to link your wallet

File: centrifuge-app/src/utils/useAuth.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { useQuery } from 'react-query'
```
Comment (jpangelle) [inline centrifuge-app/src/utils/useAuth.tsx:1]: custom hook to check/confirm auth status

File: centrifuge-js/src/modules/auth.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import { Keyring } from '@polkadot/keyring'
```
Comment (jpangelle) [inline centrifuge-js/src/modules/auth.ts:1]: new module in `centrifuge-js` for auth utilities

File: centrifuge-js/src/modules/pod.ts [hunk 4]
```diff
@@ -1,8 +1,5 @@
-import { Signer } from '@polkadot/api/types'
```
Comment (jpangelle) [inline centrifuge-js/src/modules/pod.ts:1]: migrated `signToken` function to `auth` module

File: fabric/src/components/Stepper/index.tsx [hunk 5]
```diff
@@ -52,19 +66,32 @@ export const Step = (props: StepProps & EnrichedStepProps) => {
...
+  return children.reduce<React.ReactNode[]>((acc, child) => {
+    if (React.isValidElement(child) && child.props.children) {
+      return [...acc, ...flattenReactChildren(child.props.children)]
+    }
+
+    return [...acc, child]
+  }, [])
+}
+
 export const Stepper = (props: StepperProps) => {
```
Comment (jpangelle) [inline fabric/src/components/Stepper/index.tsx:81]: added support for nested React components

File: centrifuge-app/package.json [hunk 6]
```diff
@@ -47,7 +47,8 @@
     "rxjs": "^7.5.6",
     "styled-components": "^5.3.5",
     "styled-system": "^5.1.5",
-    "typescript": "^4.7.4"
+    "typescript": "^4.7.4",
+    "universal-cookie": "^4.0.4"
```
Comment (jpangelle) [inline centrifuge-app/package.json:51]: installed a cookie parsing library

File: centrifuge-app/src/pages/Onboarding/InvestorType.tsx [hunk 7]
```diff
@@ -10,88 +10,97 @@ const examplePool = {
...
 export const InvestorType = ({
   investorType,
   isAgreedToDataSharingAgreement,
+  nextStep,
   setInvestorType,
   setIsAgreedToDataSharingAgreement,
 }: Props) => {
   const [isDataSharingAgreementDialogOpen, setIsDataSharingAgreementDialogOpen] = useState(false)
 
   return (
```
Comment (jpangelle) [inline centrifuge-app/src/pages/Onboarding/InvestorType.tsx:27]: huge diff but all I did was wrap the component in an empty fragment and add an adjacent button

File: centrifuge-app/src/pages/Onboarding/index.tsx [hunk 8]
```diff
@@ -1,23 +1,41 @@
-import { Box, Button, Flex, IconX, Shelf, Stack } from '@centrifuge/fabric'
```
Comment (jpangelle) [inline centrifuge-app/src/pages/Onboarding/index.tsx:1]: - integrates `LinkWallet` component
- conditionally renders steps based on user's investor type

File: centrifuge-app/src/components/Web3Provider.tsx [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import { isWeb3Injected } from '@polkadot/extension-dapp'
```
Comment (jpangelle) [inline centrifuge-app/src/components/Web3Provider.tsx:1]: created a loading state for when wallet status is being fetched from browser

File: centrifuge-app/src/components/PodAuthProvider.tsx [hunk 10]
```diff
@@ -18,7 +20,7 @@ function getPersisted() {
 
 export const PodAuthProvider: React.FC = ({ children }) => {
```
Comment (jpangelle) [inline centrifuge-app/src/components/PodAuthProvider.tsx:21]: @onnovisser You will need to test if this functionality still works with these refactors.

File: centrifuge-app/src/components/PodAuthProvider.tsx [hunk 11]
```diff
@@ -43,14 +45,23 @@ export const PodAuthProvider: React.FC = ({ children }) => {
...
+        const { delegator, types } = proxy
+
+        const isAuthorizedProxy = AUTHORIZED_POD_PROXY_TYPES.some((proxyType) => types.includes(proxyType))
+
+        if (isAuthorizedProxy) {
+          setTokens((prev) => ({ ...prev, [`${address}-${delegator}`]: { signed: token, payload } }))
+        }
+      } else {
+        setTokens((prev) => ({ ...prev, [address]: { signed: token, payload } }))
+      }
```
Comment (jpangelle) [inline centrifuge-app/src/components/PodAuthProvider.tsx:62]: Does an `else` here make sense? Like is it possible for someone to login without a proxy?

File: centrifuge-app/src/components/PodAuthProvider.tsx (hunk 10 above)
Comment (onnovisser) [reply centrifuge-app/src/components/PodAuthProvider.tsx:21]: It seems to be missing the `onBehalfOf` and `proxyType` arguments now in `generateJw3t`

File: centrifuge-app/package.json (hunk 6 above)
Comment (offerijns) [reply centrifuge-app/package.json]: Maybe we can write this ourselves? Seems relatively simple and I think it's good practice to try to keep the number of external dependencies low, especially for simple utilities.

File: centrifuge-app/src/components/Web3Provider.tsx [hunk 12]
```diff
@@ -157,6 +159,22 @@ export const Web3Provider: React.FC = ({ children }) => {
     }
   }
 
+  async function getWalletStatus() {
+    try {
+      await web3Enable('centrifuge-app')
```
Comment (onnovisser) [inline centrifuge-app/src/components/Web3Provider.tsx:164]: The reason we use `@subwallet/wallet-connect` is that when a user has multiple wallets and you call `web3Enable` it's going to open all wallet extensions, which can be annoying. Our new wallet menu gives the user the option for which wallet they want to connect and only then does it show the popup. using `web3Enable` on page load negates that.

File: centrifuge-js/src/modules/auth.ts [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+import { Signer } from '@polkadot/types/types'
+import * as jw3t from 'jw3t'
+
+export function getAuthModule() {
+  async function generateJw3t(
+    address: string,
+    signer: Signer,
+    options: {
+      [key: string]: string
```
Comment (onnovisser) [inline centrifuge-js/src/modules/auth.ts:10]: `key` could use a better type. e.g. `[key in 'expiresAt' | 'onBehalfOf' | 'proxyType' | ...]: string`

File: centrifuge-app/src/components/Web3Provider.tsx (hunk 12 above)
Comment (jpangelle) [reply centrifuge-app/src/components/Web3Provider.tsx:164]: Ah ok, is there anyway to get some kind of loading indicator while the browser is checking to see if a wallet is connected? `accounts` seems to start as `null` and just stays as `null` if there are no wallets. I want to show a loading spinner while that is being checked.

**Pull request closed without merging**